test/pythonml/test_workflow_integration.py

```diff
@@ -0,0 +1,175 @@
+import unittest
+import os, shutil, sys
+import re
+import subprocess
+import functools
+import yaml
+from parameterized import parameterized
+
+with open("integration_configuration.yaml", "r") as inp:
+    configuration = yaml.safe_load(inp)
+
+unit_shortnames = configuration["unit_shortnames"]
+
+# ToDo: Refactor data clumps below into classes
+
+# Figure out paths
+asset_path = configuration["asset_path"]
+fixtures_path = configuration["fixtures"]["path"]
+settings_filename = configuration["fixtures"]["settings"]
+
+
+# Get the path to the training data
+def get_dataset_filenames(config):
+    training_set_name = config["training_set_name"]
+    predict_set_name = config["predict_set_name"]
+    return training_set_name, predict_set_name
+
+
+regression_training_file, regression_predict_file = get_dataset_filenames(
+    configuration["fixtures"]["regression"])
+classification_training_file, classification_predict_file = get_dataset_filenames(
+    configuration["fixtures"]["classification"])
+clustering_training_file, clustering_predict_file = get_dataset_filenames(
+    configuration["fixtures"]["clustering"])
+
+# Which files should we remove
+files_to_remove = configuration["files_to_remove"]
+extensions_to_remove = configuration["extensions_to_remove"]
+
+
+# Extract the list of tests
+def get_test_names_configs(configuration):
+    names = [i[0] for i in configuration]
+    tests = [i[1]["units_to_run"] for i in configuration]
+    return tests, names
+
+
+all_tests = configuration["tests"]
+
+regression_configs = list(filter(lambda i: i[1]["category"] == "regression", all_tests.items()))
+tests_regression, names_regression = get_test_names_configs(regression_configs)
+
+classification_configs = list(filter(lambda i: i[1]["category"] == "classification", all_tests.items()))
+tests_classification, names_classification = get_test_names_configs(classification_configs)
+
+clustering_configs = list(filter(lambda i: i[1]["category"] == "clustering", all_tests.items()))
+tests_clustering, names_clustering = get_test_names_configs(clustering_configs)
+
+
+def custom_name_func(test_names):
+    def inner(testcase_func, param_num, param):
+        config_name = list(test_names)[int(param_num)]
+        return "%s_%s" % (
+            testcase_func.__name__,
+            parameterized.to_safe_name(config_name),)
+
+    return inner
+
+
+class BasePythonMLTest(unittest.TestCase):
+    category = ""
+
+    def setUp(self) -> None:
+        with open(os.path.join(fixtures_path, settings_filename), "r") as inp, open(settings_filename, "w") as outp:
+            for line in inp:
+                line = re.sub("PROBLEM_CATEGORY_HERE", self.category, line)
```

Comment:
Problem Category?  Given that we can't see the LFS files in a code review, can we have a comment here explaining what's going on?

Reply:
Added a comment explaining this